primitives/src/host_functions/mod.rs

```diff
@@ -0,0 +1,40 @@
+// We use such inconvienient representation since in this form passes
+// wasm-native boundary without conversion.
+// We could use sth more ergonomic but it would require encoding and decoding.
+// It's possible to pass some wrapper around Fr by implementing PassBy
+pub type Input = (u64, u64, u64, u64);
+pub type Output = (u64, u64, u64, u64);
+
+fn field_element(input: Input) -> ark_poseidon::Fr {
+    let input: [u64; 4] = [input.0, input.1, input.2, input.3];
+    ark_poseidon::Fr::new(ark_poseidon::BigInteger256::new(input))
+}
+
+fn output(field_element: ark_poseidon::Fr) -> Output {
+    let a = field_element.0 .0;
+    (a[0], a[1], a[2], a[3])
+}
+
+#[sp_runtime_interface::runtime_interface]
+pub trait Poseidon {
+    fn hash_one_to_one(input: Input) -> Output {
```

Comment:
`x_to_one_hash`? so that we stay consistent with both `poseidon` lib and wasm counterparts

Reply:
done